admin/class-botamp-admin.php

```diff
@@ -361,6 +404,241 @@ public function entity_url_cb() {
+
+		echo '<div id="notifications"><h3>' . __( 'Notifications' ) . '</h3>';
+		echo "<script>
+			 	window.fbAsyncInit = function() {
+				    FB.init({
+				      appId      : '{$page_attributes['facebook_app_id']}',
+				      xfbml      : true,
+				      version    : 'v2.6'
+				    });
+				};
+
+			  	(function(d, s, id){
+				    var js, fjs = d.getElementsByTagName(s)[0];
+				      if (d.getElementById(id)) {return;}
+				      js = d.createElement(s); js.id = id;
+				      js.src = '//connect.facebook.net/en_US/sdk.js';
+				      fjs.parentNode.insertBefore(js, fjs);
+				  }(document, 'script', 'facebook-jssdk')
+			  	);
+			</script>
+			<div class='fb-send-to-messenger'
+			  messenger_app_id='{$page_attributes['facebook_app_id']}'
+			  page_id='{$page_attributes['facebook_id']}'
+			  data-ref='$ref'
+			  color='blue'
+			  size='standard'>
+			</div></div>";
+
+	}
+
+	public function after_checkout( $order_id ) {
+		$contact = $this->botamp->contacts->get( $_POST['botamp_contact_ref'] );
+		if ( false === $contact ) {
+			return;
+		}
+
+		$entity = $this->create_entity( $order_id );
+
+		$subscription = $this->create_subscription( $entity, $contact );
+
+		add_post_meta( $order_id, 'botamp_subscription_id', $subscription->getBody()['data']['id'] );
+	}
+
+	public function add_query_vars( $vars ) {
+		$vars[] = 'botamp_order_unsubscribe';
+
+			return $vars;
+	}
+
+	public function add_unsubscribe_button( $actions, $order ) {
+		$subscription_id = get_post_meta( $order->id, 'botamp_subscription_id', true );
+
+		if ( empty( $subscription_id ) ) {
+			return;
+		}
+
+		$actions['botamp_unsubscribe_button'] = [
+			'name' => __( 'Unsubscribe from notifications', 'botamp' ),
+			'url' => $this->unsubscribe_endpoint_url( $order->id ),
+		];
+
+		return $actions;
+	}
+
+	public function add_unsubscribe_all_button( $has_orders ) {
+		if ( ! $has_orders || $this->orders_subscribed() === false ) {
+			return;
+		}
+
+		$url = $this->unsubscribe_endpoint_url( 'all' );
+
+		echo "<div id='botamp_unsubscribe_container'><a href= '{$url}' class='button botamp_unsubscribe_button'>" .
+			__( 'Unsubscribe from all your order notifications', 'botamp' ) .
+			'</a></div>';
+	}
+
+	public function update_entity( $order_id ) {
+		$subscription_id = get_post_meta( $order_id, 'botamp_subscription_id', true );
+
+		if ( empty( $subscription_id ) ) {
+			return;
+		}
+
+		$order = new WC_Order( $order_id );
+
+		$subscription = $this->botamp->subscriptions->get( $subscription_id );
+		$entity_id = $subscription->getBody()['data']['attributes']['entity_id'];
+		$entity = $this->botamp->entities->get( $entity_id );
+
+		$entity_attributes = $entity->getBody()['data']['attributes'];
+		$entity_attributes['status'] = $order->get_status();
+		$entity_attributes['meta'] = $this->get_order_meta( $order );
+
+		$this->botamp->entities->update( $entity_id, $entity_attributes );
+	}
```

Comment:
Can you move this code to an entity class or something, same for create, etc. And subscription stuff to a subscription class and so on for other types of data?